ansible_base/lib/abstract_models/common.py

```diff
@@ -123,7 +127,7 @@ def save(self, *args, **kwargs):
           * It will automatically add a created_by fields for new items
         '''
 
-        if not self.pk:
+        if not self.pk and (not hasattr(self, 'username') or self.username != get_system_username()[0]):
```

Comment:
This condition is too hard to read. I suggest introducing an instance method `is_the_system_user` which returns `hasattr(self, 'username') and self.username == get_system_username()[0]`

Reply:
I realize this is a really base-level abstract model class. If you didn't want this method to be available on almost all instances, then add it as a static method at the top of this file which would take a `model.Model` as its argument.

Reply:
How would you feel if I modified lib.utils.models.is_system_user to take `model.Model` and then add a check for "Is this model a user"? That way there is just one function in the system for this.

Reply:
Yeah, I saw there was a very similar method in that module. It would make sense to reuse that somewhere in the call stack for this. However it shakes out should be fine.